src/lightly_train/_commands/export_task.py

```diff
@@ -165,13 +185,19 @@ def _export_task_from_config(config: ExportTaskConfig) -> None:
 
             onnx.checker.check_model(out_path, full_check=True)
 
-            x = torch.rand_like(dummy_input)
+            # Always run the reference input in float32 and on cpu for consistency
+            x_model = torch.rand_like(dummy_input, dtype=torch.float32, device="cpu")
+            x_onnx = x_model.to(onnx_dtype)
+
             session = ort.InferenceSession(out_path)
-            input_feed = {input_name: x.cpu().numpy()}
+            input_feed = {input_name: x_onnx.numpy()}
             outputs_onnx = session.run(output_names=output_names, input_feed=input_feed)
             outputs_onnx = tuple(torch.from_numpy(y) for y in outputs_onnx)
 
-            outputs_model = task_model(x)
+            task_model = task_model_helpers.load_model_from_checkpoint(
+                checkpoint=checkpoint_path, device="cpu"
+            )
+            outputs_model = task_model(x_model)
```

Comment:
Why not `task_model.to("cpu")` instead of loading again from the checkpoint?

Reply:
My idea is here that there might be some degradation in quality when we call ` task_model.to(onnx_dtype)` above. Therefore I want to rather compare the output of the model that has not been converted to another dtype the onnx model that has been converted. That is also why call `x_model = torch.rand_like(dummy_input, dtype=torch.float32, device="cpu")` above.
`

Reply:
Makes sense 👍🏼